modin/test/storage_formats/pandas/test_internals.py

```diff
@@ -110,3 +110,79 @@ def func_to_apply(partition, row_internal_indices, col_internal_indices, item):
     md_df._query_compiler._modin_frame = new_modin_frame
 
     df_equals(md_df, pd_df)
+
+
+small_dfs = [
+    pd.DataFrame(
+        [[i + j for j in range(0, 1000)]],
+        columns=[f"col{j}" for j in range(1, 1001)],
+        index=pd.Index([i - 1]),
+    )
+    for i in range(1, 100001, 1000)
+]
+large_df = pd.DataFrame(
+    [[i + j for j in range(1, 1000)] for i in range(0, 100000, 1000)],
+    columns=[f"col{j}" for j in range(1, 1000)],
+    index=pd.Index(list(range(0, 100000, 1000))),
+)
+
+
+@pytest.mark.skipif(
+    Engine.get() not in ("Dask", "Ray"),
+    reason="Rebalancing partitions is only supported for Dask and Ray engines",
+)
+@pytest.mark.parametrize(
+    "large_df,col_length",
+    [
+        (pd.concat(small_dfs), 100),
+        (pd.concat([pd.concat(small_dfs)] + small_dfs[:3]), 103),
+        (pd.concat([large_df] + small_dfs[:3]), 103),
```

Comment:
I don't think we should call the Modin function under test within the pytest params. If the Modin `concat` throws an unexpected exception (not that unlikely when someone is modifying modin), I think that pytest will fail while gathering the tests, and the developer will probably be confused. I understand that there is some repeated work in these test cases, and you want to test rebalancing different kinds of partitions, but could you make it more succinct in some other way? What about using labels like `many_small_dfs`, `rebalance_twice`, and `large_partitions_with_smaller_partitions`, and doing the concat that corresponds to the label?